components/email_aliases/email_aliases_service_unittest.cc

```diff
@@ -0,0 +1,233 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/email_aliases/email_aliases_service.h"
+
+#include <string>
+#include <utility>
+#include <vector>
+
+#include "base/functional/bind.h"
+#include "base/test/bind.h"
+#include "base/test/scoped_feature_list.h"
+#include "base/test/task_environment.h"
+#include "brave/components/email_aliases/features.h"
+#include "components/grit/brave_components_strings.h"
+#include "services/network/public/cpp/weak_wrapper_shared_url_loader_factory.h"
+#include "services/network/test/test_url_loader_factory.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "ui/base/l10n/l10n_util.h"
+
+namespace email_aliases {
+
+using AuthenticationStatus = email_aliases::mojom::AuthenticationStatus;
+
+namespace {
+
+// Test observer for authentication state changes
+class TestObserver : public email_aliases::mojom::EmailAliasesServiceObserver {
+ public:
+  void OnAuthStateChanged(email_aliases::mojom::AuthStatePtr state) override {
+    last_state = state->status;
+    if (last_state == expected_status_) {
+      run_loop_->Quit();
+    }
+  }
+  void StartRunLoop(AuthenticationStatus expected_status) {
+    run_loop_ = std::make_unique<base::RunLoop>();
+    expected_status_ = expected_status;
+  }
+  void WaitForExpectedStatus() {
+    run_loop_->Run();
```

Comment:
this can be simplified with base::test::RunUntil.

Reply:
done